tools/clang/test/HLSLFileCheck/dxil/intrinsics/launch_types_broadcast.ll

```diff
@@ -0,0 +1,50 @@
+; RUN: %dxv %s 
+
+; There should be no errors!
+
+target datalayout = "e-m:e-p:32:32-i1:32-i8:32-i16:32-i32:32-i64:64-f16:32-f32:32-f64:64-n8:16:32:64"
+target triple = "dxil-ms-dx"
+
+define void @entry2() {
+  %1 = call i32 @dx.op.threadId.i32(i32 93, i32 1)
+  %2 = call i32 @dx.op.groupId.i32(i32 94, i32 2)
+  %3 = call i32 @dx.op.threadIdInGroup.i32(i32 95, i32 1)
+  %4 = call i32 @dx.op.flattenedThreadIdInGroup.i32(i32 96)
+  ret void
+}
+
+!llvm.ident = !{!0}
+!dx.version = !{!1}
+!dx.valver = !{!2}
+!dx.shaderModel = !{!3}
+!dx.typeAnnotations = !{!4}
+!dx.entryPoints = !{!8, !9}
+
+!0 = !{!"dxc(private) 1.7.0.14317 (main, e3c311409675)"}
+!1 = !{i32 1, i32 3}
+!2 = !{i32 1, i32 8}
+!3 = !{!"lib", i32 6, i32 3}
+!4 = !{i32 1, void ()* @entry2, !5}
+!5 = !{!6}
+!6 = !{i32 0, !7, !7}
+!7 = !{}
+!8 = !{null, !"", null, null, null}
+!9 = !{void ()* @entry2, !"entry2", null, null, !10}
+!10 = !{i32 8, i32 15, i32 13, i32 1, i32 15, !11, i32 16, i32 -1, i32 18, !12, i32 4, !12, i32 5, !13}
+!11 = !{!"entry2", i32 0}
+!12 = !{i32 1, i32 1, i32 1}
+!13 = !{i32 0}
+
+; Function Attrs: nounwind readnone
+declare i32 @dx.op.threadId.i32(i32, i32) #1
+
+; Function Attrs: nounwind readnone
+declare i32 @dx.op.groupId.i32(i32, i32) #1
+
+; Function Attrs: nounwind readnone
+declare i32 @dx.op.threadIdInGroup.i32(i32, i32) #0
+
+; Function Attrs: nounwind readnone
+declare i32 @dx.op.flattenedThreadIdInGroup.i32(i32) #0
+
+; this file comes from compiling launch_types.hlsl with:
```

Comment:
This is incomplete, and I don't think there's a `launch_types.hlsl` currently in the project.

Also, can we move lines like this to the top so generated/modified code follows any informative comments like this?

Reply:
@bob80905, this too.

Reply:
I resolved this by removing the comment entirely. When I get around to implementing these diagnostics in Sema, then I will add launch_types.hlsl, and add a reference in these .ll files to refer back to that file. But, it doesn't make sense to have this reference in this PR because the file isn't being added yet. It can't be added because the test would fail, since diagnostic implementation in Sema should happen separately. launch_types.hlsl tests each launch type for sema diagnostics, and if the sema implementation happens, then validation code won't get to execute, so these files need to go in first without any knowledge of launch_types.hlsl